### DISCUSSION TITLE
Compiled Query and Clean Architecture

### DISCUSSION THREAD

**Discussion by gokujax** [jasontaylordev/CleanArchitecture#846] [Q&A] [answered]
I'd like to create compiled queries for the ones i commonly use or call in some commands and queries.
and I can only access IApplicationDbContext interface not ApplicationDbContext in the Application layer to create the compiled query methods.
Please help

**Comment by crazyfx1** [comment, marked as answer]
You could either create a Get method in the IApplicationDbContext which fetches the data using a compiled query in the ApplicationDbContext implementation.

Or you could create a method / property in the IApplicationDbContext which returns the already instantiated compiled query enumerable.
In ApplicationDbContext:
```
public IEnumerable<TodoItem> TodoItemQuery => CompiledQuery(this);
```

**Reply by gokujax** [reply]
Thanks @crazyfx1
